src/ai/backend/manager/models/container_registry.py

```diff
@@ -0,0 +1,425 @@
+        "is_global", sa.Boolean, nullable=True, server_default=sa.text("true"), index=True
+    )
+
+    @classmethod
+    async def get(
+        cls,
+        session: AsyncSession,
+        id: str | uuid.UUID,
+    ) -> ContainerRegistryRow:
+        query = sa.select(ContainerRegistryRow).where(ContainerRegistryRow.id == id)
+        result = await session.execute(query)
+        row = result.scalar()
+        if row is None:
+            raise NoResultFound
+        return row
+
+    @classmethod
+    async def list_by_registry_name(
+        cls,
+        session: AsyncSession,
+        registry_name: str,
+    ) -> Sequence[ContainerRegistryRow]:
+        query = sa.select(ContainerRegistryRow).where(
+            ContainerRegistryRow.registry_name == registry_name
+        )
+        result = await session.execute(query)
+        rows = result.scalars().all()
+        if not rows:
+            raise NoResultFound
+        return rows
+
+
+class CreateContainerRegistryInput(graphene.InputObjectType):
+    url = graphene.String(required=True, description="Added in 24.09.0.")
+    type = ContainerRegistryTypeField(
+        required=True,
+        description=f"Registry type. One of {ContainerRegistryTypeField.allowed_values}. Added in 24.09.0.",
+    )
+    registry_name = graphene.String(required=True, description="Added in 24.09.0.")
+    is_global = graphene.Boolean(description="Added in 24.09.0.")
+    project = graphene.String(description="Added in 24.09.0.")
+    username = graphene.String(description="Added in 24.09.0.")
+    password = graphene.String(description="Added in 24.09.0.")
+    ssl_verify = graphene.Boolean(description="Added in 24.09.0.")
+
+
+class ModifyContainerRegistryInput(graphene.InputObjectType):
+    id = graphene.String(
+        required=True,
+        description="Object id. Can be either global id or object id. Added in 24.09.0.",
+    )
+    url = graphene.String(description="Added in 24.09.0.")
+    type = ContainerRegistryTypeField(
+        description=f"Registry type. One of {ContainerRegistryTypeField.allowed_values}. Added in 24.09.0."
+    )
+    registry_name = graphene.String(description="Added in 24.09.0.")
+    is_global = graphene.Boolean(description="Added in 24.09.0.")
+    project = graphene.String(description="Added in 24.09.0.")
+    username = graphene.String(description="Added in 24.09.0.")
+    password = graphene.String(description="Added in 24.09.0.")
+    ssl_verify = graphene.Boolean(description="Added in 24.09.0.")
+
+
+class DeleteContainerRegistryInput(graphene.InputObjectType):
+    """Added in 24.09.0."""
+
+    id = graphene.String(
+        required=True,
+        description="Object id. Can be either global id or object id. Added in 24.09.0.",
+    )
+
+
+class ContainerRegistryConfig(graphene.ObjectType):
+    url = graphene.String(required=True, description="Added in 24.09.0.")
+    type = ContainerRegistryTypeField(required=True, description="Added in 24.09.0.")
```

Comment:
As gql inspector checked, `type` field changed from `string` to `ContainerRegistryTypeField`.
We can deprecate this field and add a new `ContainerRegistryTypeField` field.